llvm/lib/Target/RISCV/RISCVTargetTransformInfo.cpp

```diff
@@ -1163,7 +1165,7 @@ RISCVTTIImpl::getIntrinsicInstrCost(const IntrinsicCostAttributes &ICA,
     auto LT = getTypeLegalizationCost(RetTy);
     if (const auto *Entry = CostTableLookup(VectorIntrinsicCostTable,
                                             ICA.getID(), LT.second))
-      return LT.first * Entry->Cost;
+      return LT.first * TLI->getLMULCost(LT.second) * Entry->Cost;
```

Comment:
This line is incorrect.  The LMUL is already considered in the entry in the cost table.

Reply:
The entries in the cost table seem to be the same across various LMULs though, am I missing something here?
E.g. nxv1i64 is LMUL1, nxv2i64 is LMUL2 etc.
```c++
    {Intrinsic::bswap, MVT::nxv1i64, 31},
    {Intrinsic::bswap, MVT::nxv2i64, 31},
    {Intrinsic::bswap, MVT::nxv4i64, 31},
    {Intrinsic::bswap, MVT::nxv8i64, 31},
```

Reply:
I guess we can't exactly tell what percentage of instructions that make up the cost actually need to be scaled by LMUL. Would it be safe to say though that most of them are vector instructions that read and write full vector groups?

Reply:
The cost table may be *wrong* and need updating, but the structure already accounts for LMUL scaling.  We could also rethink the table (i.e. have a table for linearly scaled cases, and one for all others), but that's different from the way this patch approaches it.

Reply:
p.s. For forward progress, I would suggest this patch just drop the intrinsic table case and we handle that in a separate PR.

Reply:
Thinking about this further.  The actual values in the table aren't LMUL sensitive.  It does depend on element type (i.e. SEW).  I went ahead and modified the table (8bf624af4776cdaad82c0fc4da9e9b0f14e9ea10) to make that explicit.  

When doing that, I realized I'd been completely wrong here the whole time.  The original table was keyed by vector type, and that's fixed with respect to LMUL.  What got me confused is that changing VLEN (a related but distinct notion), does change the fixed mappings.  I got myself confused, sorry for the confusion!

Given that, LGTM to the change as whole.